test/functional/api/cas/casadm.py

```diff
@@ -1,398 +1,519 @@
+def stop_cache(cache_id: int, no_data_flush: bool = False, shortcut: bool = False) -> Output:
     output = TestRun.executor.run(
-        stop_cmd(cache_id=str(cache_id), no_data_flush=no_data_flush, shortcut=shortcut))
+        stop_cmd(cache_id=str(cache_id), no_data_flush=no_data_flush, shortcut=shortcut)
+    )
     if output.exit_code != 0:
         raise CmdException("Failed to stop cache.", output)
     return output
 
 
-def add_core(cache: Cache, core_dev: Device, core_id: int = None, shortcut: bool = False):
+def set_param_cutoff(
+    cache_id: int,
+    core_id: int = None,
+    threshold: Size = None,
+    policy: SeqCutOffPolicy = None,
+    promotion_count: int = None,
+    shortcut: bool = False,
+) -> Output:
     _core_id = None if core_id is None else str(core_id)
-    output = TestRun.executor.run(
-        add_core_cmd(cache_id=str(cache.cache_id), core_dev=core_dev.path,
-                     core_id=_core_id, shortcut=shortcut))
+    _threshold = None if threshold is None else str(int(threshold.get_value(Unit.KibiByte)))
+    _policy = None if policy is None else policy.name
+    _promotion_count = None if promotion_count is None else str(promotion_count)
+    command = set_param_cutoff_cmd(
+        cache_id=str(cache_id),
+        core_id=_core_id,
+        threshold=_threshold,
+        policy=_policy,
+        promotion_count=_promotion_count,
+        shortcut=shortcut,
+    )
+    output = TestRun.executor.run(command)
     if output.exit_code != 0:
-        raise CmdException("Failed to add core.", output)
-    core = Core(core_dev.path, cache.cache_id)
-    return core
+        raise CmdException("Error while setting sequential cut-off params.", output)
+    return output
 
 
-def remove_core(cache_id: int, core_id: int, force: bool = False, shortcut: bool = False):
+def set_param_cleaning(cache_id: int, policy: CleaningPolicy, shortcut: bool = False) -> Output:
     output = TestRun.executor.run(
-        remove_core_cmd(cache_id=str(cache_id), core_id=str(core_id),
-                        force=force, shortcut=shortcut))
+        set_param_cleaning_cmd(cache_id=str(cache_id), policy=policy.name, shortcut=shortcut)
+    )
     if output.exit_code != 0:
-        raise CmdException("Failed to remove core.", output)
+        raise CmdException("Error while setting cleaning policy.", output)
+    return output
 
 
-def remove_inactive(cache_id: int, core_id: int, force: bool = False, shortcut: bool = False):
+def set_param_cleaning_alru(
+    cache_id: int,
+    wake_up: int = None,
+    staleness_time: int = None,
+    flush_max_buffers: int = None,
+    activity_threshold: int = None,
+    shortcut: bool = False,
+) -> Output:
+    _wake_up = str(wake_up) if wake_up else None
+    _staleness_time = str(staleness_time) if staleness_time else None
+    _flush_max_buffers = str(flush_max_buffers) if flush_max_buffers else None
+    _activity_threshold = str(activity_threshold) if activity_threshold else None
```

Comment:
wake_up and activity_threshold can be set to 0, so you need to do the check with `is not None`  and actually you probably should do it for all ints that can be None in the whole file (so casadm can reject 0 instead of masking it as empty value)